tests/src/test/java/alluxio/server/configuration/ConfigCheckerIntegrationTest.java

```diff
@@ -0,0 +1,150 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.server.configuration;
+
+import alluxio.Configuration;
+import alluxio.Constants;
+import alluxio.PropertyKey;
+import alluxio.PropertyKey.Scope;
+import alluxio.client.MetaMasterClient;
+import alluxio.multi.process.MultiProcessCluster;
+import alluxio.testutils.BaseIntegrationTest;
+import alluxio.wire.MasterInfo;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+/**
+ * Test server-side configuration checker.
+ */
+public class ConfigCheckerIntegrationTest extends BaseIntegrationTest {
+  private static final int TEST_NUM_MASTERS = 2;
+  private static final int TEST_NUM_WORKERS = 2;
+
+  @Test
+  public void MultiMasters() throws Exception {
+    Map<Integer, Map<PropertyKey, String>> masterProperties
+        = generateProperties(TEST_NUM_MASTERS, Scope.MASTER);
```

Comment:
I think this would be simpler to understand if we just hardcode 2 different master configurations here.

Reply:
Then for all the test, we need to hardcode. I think the current way may be better if we want to change the num of masters and num of workers to test.